src/utils/progress_utils.py

```
@@ -0,0 +1,207 @@
+            current_progress = self.start_progress + int(item_progress)
+        else:
+            current_progress = self.end_progress
+        
+        message = f"Processed {self.processed_items}/{self.total_items}"
+        if item_name:
+            message += f" - {item_name}"
+            
+        await self.report(current_progress, message)
+
+
+async def with_progress_reporting(ctx: Context, tool_name: str, operation_func, *args, **kwargs):
```

Ruff (0.12.2)

Missing return type annotation for public function `with_progress_reporting`

(ANN201)